common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -4,31 +4,40 @@ module Dependabot
   module Config
     # Filters versions that should not be considered for dependency updates
     class IgnoreCondition
-      UPDATE_TYPES = %i(
-        ignore_major_versions
-        ignore_minor_versions
-        ignore_patch_versions
-      ).freeze
+      UPDATE_TYPE_LOOKUP = {
+        "version-update:semver-patch" => :ignore_patch_versions,
+        "version-update:semver-minor" => :ignore_minor_versions,
+        "version-update:semver-major" => :ignore_major_versions
+      }.freeze
```

Comment:
Question: what does this remapping accomplish now?

If we're happy assigning meaning to these strings, mapping them to string-agnostic symbols is a little moot.

Reply:
Ah yes good point, will rejig this. I didn't think of persisting a transformed value but seems easier to just pass the string along as is? Typed enum would have been real nice!